workos/_base_client.py

```diff
@@ -13,37 +17,98 @@
 from workos.webhooks import WebhooksModule
 
 
-class BaseClient(Protocol):
+HTTPClientType = TypeVar("HTTPClientType", bound=HTTPClient)
+
+
+class BaseClient(Generic[HTTPClientType]):
```

Comment:
Did a bit of a rework here to pass in the HTTP client type, partly because Python versions <3.11 were having issues resolving the _http_client attribute properly.